### DISCUSSION TITLE
pex command failing with version error for sendgrid on macos

### DISCUSSION THREAD

**Discussion by suriya92** [pex-tool/pex#1529] [Q&A]
Hi There, 
pex command is failing on macos(version:11.6,intel chip)with the following error when attempted to include sendgrid(target platform is linux)

### ERROR: Could not find a version that satisfies the requirement starkbank-ecdsa>=2.0.1 (from sendgrid)
ERROR: No matching distribution found for starkbank-ecdsa>=2.0.1

pex command: **pex.venv/bin/pex -vvvvvvvv --output-file output.zip  --manylinux=manylinux2014 --resolve-local-platforms --no-pypi --index=https://pypi.org/simple/ --resolver-version pip-2020-resolver --platform linux-x86_64-cp-38-cp38 "sendgrid==6.9.1"**
The same command works fine on Linux. Appreciate any help on resolving this.
Thanks !

**Comment by jsirois** [comment]
This was answered in #1530, but the short of it is: whenever you specify `--platform` and that platform is foreign (you're trying to resolve for Linux on a Mac), every distribution must be available as a pre-built wheel. In this case the `starkbank-ecdsa` distribution is only available as a source distribution for versions 2.0.1, 2.0.2 and 2.0.3. Since Pex does not know how to cross-build a Linux wheel on a Mac machine, you get this error. To work around the error you need to:
1. Get access to a Linux machine (a vm or Docker image is fine).
2. Build a wheel for the `starkbank-ecdsa` sdist.
3. Add the directory contining the wheel to your resolve with `--find-links /custom/wheel/directory`
